File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.ts
function findMin(nums: number[]): number {
    let ans = Infinity;
    let l = 0;
    let r = nums.length - 1;

    while (l <= r) {
        let m = Math.floor((l + r) / 2);
        if (nums[l] <= nums[m]) {  // left is sorted
            ans = Math.min(ans, nums[l]);
            l = m + 1;
        }
        else {  // right is sorted
            ans = Math.min(ans, nums[m]);
            r = m - 1;
        }
    }

    return ans;
};

File: 153-find-minimum-in-rotated-sorted-array/find-minimum-in-rotated-sorted-array.java
class Solution {
    public int findMin(int[] nums) {
        int l = 0;
        int r = nums.length - 1;
        int ans = Integer.MAX_VALUE;
        while(l <= r) {
            int m = l + (r - l) / 2;
            if(nums[l] <= nums[m]) {
                ans = Math.min(ans , nums[l]);
                l = m + 1;
            }
            else {
                ans = Math.min(ans , nums[m]);
                r = m - 1;
            }
        }
        return ans;
    }
}